src/thing_effects.c

```diff
@@ -282,6 +282,44 @@ struct Thing *create_effect_element(const struct Coord3d *pos, unsigned short ee
     return thing;
 }
 
+// Initial state from the side view of the sphere:
+// 'r' is the radius of the circle
+// The equation of the circle is x^2 + y^2 = r^2
+// Solving for x, we get x = sqrt(r^2 - y^2)
+// As 'y' varies, we recompute 'x', which is the new radius for the lower and upper parts of the sphere
+// The point of origin would be the center of the creature
+// Starting at y = -r, then x = 0 (a single dot)
+// In 3D space, 'y' corresponds to 'z', representing the height
+void process_cleanse_effect(struct Thing* thing) {
+    struct CreatureControl* cctrl = creature_control_get_from_thing(thing);
+
+    int height = thing->clipbox_size_z + thing->floor_height + (5 * gameadd.crtr_conf.exp.size_increase_on_exp * cctrl->explevel);
```

Comment:
What is the '5' here?

Reply:
well, normally we multiply with the thing->clipbox_size_z.
So:
```c
int height = thing->clipbox_size_z + (thing->clipbox_size_z * gameadd.crtr_conf.exp.size_increase_on_exp * cctrl->explevel);
```

I'm also not so sure about including the floor height in this calculation. Why is this relevant? Also consider a flying creature might have it.

Reply:
![image](https://github.com/dkfans/keeperfx/assets/13840686/c21add34-2237-4cc0-ad98-9ac8cf91ca60)
Floor height is not a constant value.